src/devices/Pn532/Pn532.cs

```diff
@@ -26,7 +26,9 @@ namespace Iot.Device.Pn532
     /// </summary>
     public class Pn532 : CardTransceiver, IDisposable
     {
-        private const int I2cMaxBuffer = 1024;
+        // Host-controller frame size
+        private const int MaxPacketData = 264;      // maximum length of packet data
```

Comment:
Yes, this constant could be public. After all, it comes from the data sheet.

The exercise for the reader is a little different. There are several places that simply assumed a maximun response of 1024 bytes, but the actual response will be much smaller . It is almost a 4x performance improvement to use MaxPacketData instead of 1024, but it would be better still for each of these to determine the actual maximum response size and use that instead.

Reply:
Got that, but (blindly) changing it to the max size would already help, so if the constant is public,  that's very easy.

Reply:
I'm not sure I understand. I did change them from 1024 to ```MaxPacketData```. They're all within this class, but I can make the constant public in case something else wants to take advantage of it.

Reply:
If I understand correctly Patrick, it's more a property with this constant as initial value than having a public constant.

Reply:
Now the question as you've been reading all this deeply recently, is: is there an interest to change that value? Or is this 264 + 12 the absolute maximum value?

Reply:
Out of the 264 total, the first byte of the command and response payload is the command code (which is incremented by one in the response), and then for InDataExchange the next byte is the target. I think the largest such payload is a command APDU, which is CLA,INS,P1,P2,Lc,Data,Le (where Data can be up to 255 bytes). [However, that's only 261]

When the length of the packet data exceeds 255 it uses the extended information frame, which is 12 bytes. (This includes the command code that is prefixed to the payload.)

So, in my understanding, the largest payload size will always be <= 264 and the largest total size will always be <= 276. The I2C buffer in ReadResponseI2C can be sized according to the expected data. It is easiest to just add 12 (for the extended information frame). It could be more precise and use the smaller normal information frame, but the difference is small enough at that point that it wasn't worth adding the extra logic to compute that.

Reply:
Then, we don't need to expose it. And it's all ok for this optimization. Thanks again!